media-sound/fluid-soundfont/fluid-soundfont-3.1.ebuild

```diff
@@ -1,29 +1,62 @@
-# Copyright 1999-2020 Gentoo Authors
+# Copyright 1999-2021 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=7
 
-DESCRIPTION="The Fluid R3 soundfont"
-HOMEPAGE="http://musescore.org/en/handbook/soundfont"
+# There are no new releases for old data files but the archival repo can update
+# the accompanying documentation [but this ebuild only cares for the data ;) ].
+COMMIT=700c7c1775da12f630750cc34149763ee52aa6c6
+
+DESCRIPTION="The Fluid R3 version 1 soundfont"
+HOMEPAGE="https://github.com/pinkflames/fluid-soundfont"
 SRC_URI="
-	http://prereleases.musescore.org/soundfont/${PN}_${PV}.tar.gz
-	timidity? ( https://dev.gentoo.org/~hwoarang/distfiles/timidity.cfg.bz2 )"
+	https://github.com/pinkflames/${PN}/archive/${COMMIT}.tar.gz -> ${PN}-${COMMIT}.tar.gz
+"
 
 LICENSE="MIT"
 SLOT="0"
 KEYWORDS="amd64 ~ppc64 x86"
-IUSE="timidity"
+IUSE="+gm gs"
+
+BDEPEND="media-libs/flac"
+
+REQUIRED_USE="|| ( gm gs )"
```

Comment:
No one likes `REQUIRED_USE`. Not users, not arch testers. Could you simply construct a way to choose gm if both are disabled, while printing a message to the user?

Reply:
I would have thought that installing a feature `foo` while `-foo` USE flag is set, would be a very, very bad and terrible thing to do but, if that really would be acceptable, that's not a difficult thing to write, I imagine. Alternatively GM could be installed unconditionally but it makes up the vast bulk of the package size, so it feels a bit bad to foist that on someone who maybe really needs just the SFX extension.

Reply:
> I would have thought that installing a feature foo while -foo USE flag is set, would be a very, very bad and terrible thing to do but, if that really would be acceptable

It is arguable indeed. I'm doing my best to get rid of REQUIRED_USE as much as possible.

I hate to use existing ebuilds in the repository as an example, as there are some horrible ebuilds present, but check virtualbox, freeciv and qudos for example that does this already. There's many more...

> it feels a bit bad to foist that on someone who maybe really needs just the SFX extension.

Then enable `gs` if both are disabled, and do `IUSE="+gm gs"`?

Would this package be functional at all without either USE flag enabled, do we need the REQUIRED_USE?

Reply:
-gm -gs would result in effectively NOOP as only README and LICENCE would be installed. The GS while small is also, IMO, not what people expect - it's literally effect sounds only. I'm sure there's someone out there that only wants cheesy effects like clapping hands, washing waves or dog barks but majority of people who use MIDI probably will want musical instrument samples. Besides we already set +gm - the required use would only kick in if someone had -gm USE set, which itself would be weird, since it's a local USE flag and should not be set globally.

Reply:
and how about dropping the use flags and simply install both?

Reply:
That would cause confusion - how many people would know if they need the GM or the GS file? I certainly didn't when I began my work on this PR.